app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -74,23 +75,48 @@ def deploy_all
     end
   end
 
+  def create_all_stages_verify
+    @stages_preexisting, @stages_to_create = gather_stages_for_creation
+
+    render
+  end
+
   def create_all_stages
     # No more than one stage, per project, per deploy_group
     # Note: you can call this multiple times, and it will create missing stages, but no redundant stages.
+
+    _throwaway, stages_to_create = gather_stages_for_creation
+    stages_to_create.each do |template_stage|
+      create_stage_with_group(template_stage)
+    end
+
+    redirect_to [:admin, deploy_group]
+  end
+
+  private
+
+  # gather_stages_for_creation()
+  #
+  # returns a list of stages already created and list of stages to create (through their template stages)
+  def gather_stages_for_creation
     environment = deploy_group.environment
     template_stages = environment.template_stages.all
     deploy_group_stages = deploy_group.stages.all
+
+    stages_preexisting = []
+    stages_to_create = []
     Project.where(include_new_deploy_groups: true).each do |project|
       template_stage = template_stages.detect { |ts| ts.project_id == project.id }
       deploy_group_stage = deploy_group_stages.detect { |dgs| dgs.project.id == project.id }
-      if template_stage && !deploy_group_stage
-        create_stage_with_group(template_stage)
+      if deploy_group_stage
+        stages_preexisting << deploy_group_stage
+      elsif template_stage
+        stages_to_create << template_stage
       end
     end
-    redirect_to [:admin, deploy_group]
-  end
 
-  private
+    [stages_preexisting, stages_to_create]
```

Comment:
stages_preexisting sounds kind of clunky ... preexisting_stages ? ... does break the symmetry sadly ...